lib/mix/lib/mix/sync/lock.ex

```diff
@@ -97,7 +97,9 @@ defmodule Mix.Sync.Lock do
     opts = Keyword.validate!(opts, [:on_taken])
 
     hash = key |> :erlang.md5() |> Base.url_encode64(padding: false)
-    path = Path.join([System.tmp_dir!(), "mix_lock", hash])
+    base_path = Path.join([System.tmp_dir!(), "mix_lock"])
+    init_base_path(base_path)
+    path = Path.join([base_path, hash])
```

Comment:
Let's do this:

```suggestion
    path = Path.join(base_path(), hash)
```

Then we move the `base_path = Path.join([System.tmp_dir!(), "mix_lock"])` to that function :)